### ISSUE TITLE
Parameter can be confused with a section and trigger false positives

### ISSUE THREAD

**Issue by e-gebes** [PyCQA/pydocstyle#573]
I am using NumPy style, I did not check if the following also applies to Google style.

```py
def function(parameters: list, second_parameter: int):
    """Process parameters.

    Parameters
    ----------
    parameters
        Listing the parameter "parameters" here triggers 5 warnings.
    second_parameter
        This is OK.
    """
```

The 5 warnings are:

    D410 Missing blank line after section
    D414 Section has no content
    D405 Section name should be properly capitalized
    D411 Missing blank line before section
    D407 Missing dashed underline after section

Apparently pydocstyle is confused because the parameter is, apart from capitalization, spelled like a section header.

**Comment by sigmavirus24** [comment]
Duplicate of #481
